xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -810,6 +813,16 @@ void CGUIWindowManager::CloseDialogs(bool forceClose) const
   }
 }
 
+void CGUIWindowManager::CloseModalDialogs(bool forceClose) const
+{
+  CSingleLock lock(g_graphicsContext);
```

Comment:
those methods should only be called by render thread, right? if this is true protection of m_activeDialogs is not needed.
an alternative would be to introduce a separate lock for activeDialogs, gfx lock is rather heavy for that purpose